src/bundle/Controller/Content/ContentTreeController.php

```diff
@@ -140,6 +142,36 @@ public function loadSubtreeAction(Request $request): Root
         return new Root($elements);
     }
 
+    /**
+     * @param array<int> $locationIdList
+     *
+     * @return array<int, Location>
+     */
+    private function prepareLocationsArray(array $locationIdList): array
+    {
+        $locations = [];
+
+        // Always load root location using `sudo`
+        if (in_array(self::ROOT_LOCATION_ID, $locationIdList, true)) {
+            $rootLocation = $this->repository->sudo(
+                fn (): Location => $this->locationService->loadLocation(self::ROOT_LOCATION_ID)
+            );
+            $locations[$rootLocation->getId()] = $rootLocation;
+
+            $locationIdList = array_diff($locationIdList, [self::ROOT_LOCATION_ID]);
+        }
+
+        // Load rest of locations with proper permission checks
+        if ($locationIdList !== []) {
```

Comment:
```suggestion
        if (!empty($locationIdList)) {
```
maybe?

Reply:
Wouldn't `!empty` be less strict-typed?

Reply:
I prefer strict check over a loose one, especially since `empty()` === `isset() && not falsy`. Worst thing we could have is a missing variable.